lib/Sema/CSSyntacticElement.cpp

```diff
@@ -995,6 +995,10 @@ class SyntacticElementConstraintGenerator
         }
       }
 
+      if (cs.isForCodeCompletion() && !cs.containsIDEInspectionTarget(element)) {
```

Comment:
Nit: could you please a comment here with brief explanation why this is necessary?